e2e/po/app.filters.ts

```diff
@@ -0,0 +1,66 @@
+import { browser, by, element, ExpectedConditions, promise } from 'protractor';
+
+export class Filters {
+  DETERMINISTIC_WAIT = 3000; // in milliseconds
+  retryAttempts = 0;
+  EC = browser.ExpectedConditions;
+
+  /**
+   * open filter wizard
+   */
+  openFilterWizard() {
+    browser.ignoreSynchronization = true;
+    return element(by.css('.add-application a')).click();
+  }
+
+  /**
+   * Add a filter in south service pipeline
+   * @prerequisite A south service should be added in the FogLAMP.
+   * A filter should be installed and discoverable at FogLAMP side.
+   * @param filterName {string} filter name
+   * @param {string}   filter plugin name
+   */
+  addFilter(filterName: string) {
+    browser.ignoreSynchronization = true;
+    this.waitForFilterPluginsToLoad(this.DETERMINISTIC_WAIT).then(() => {
+      browser.ignoreSynchronization = true;
+      element(by.name('type')).all(by.tagName('option'))
+        .then(options => {
+          options[0].click();  // select first plugin in select box
+        });
+      // supply filter name
+      element(by.id('name')).sendKeys(filterName);
+      // click next button twice
+      element(by.id('next')).click();
+      element(by.id('next')).click();
+    })
+      .catch((error) => {
+        if (this.retryAttempts <= 5) {
+          console.log('Retrying load filter plugin.');
+          this.addFilter(filterName);
+        } else {
+          console.log('Rejecting the promise after 5 attempts.');
```

Comment:
don;t hard code 5, instead use class level variable value. We would need to have it pass through args in longer term